client/schema.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Peter Mattis (peter@cockroachlabs.com)
+
+package client
+
+import (
+	"fmt"
+	"reflect"
+	"regexp"
+	"sort"
+	"strings"
+
+	"github.com/cockroachdb/cockroach/proto"
+)
+
+var schemaOptRE = regexp.MustCompile(`\s*((?:\w|\s)+)(?:\(([^)]+)\))?\s*`)
+
+type columnsByName []proto.Column
+
+func (s columnsByName) Len() int           { return len(s) }
+func (s columnsByName) Swap(i, j int)      { s[i], s[j] = s[j], s[i] }
+func (s columnsByName) Less(i, j int) bool { return s[i].Name < s[j].Name }
+
+type indexesByName []proto.TableSchema_IndexByName
+
+func (s indexesByName) Len() int      { return len(s) }
+func (s indexesByName) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
+func (s indexesByName) Less(i, j int) bool {
+	// Sort the name "primary" less than other names.
+	if s[i].Name == "primary" {
```

Comment:
nah, i don't think it belongs in keys. `proto` is fine for now, and it'll move out to structured when the proto split happens